Source/JavaScriptCore/runtime/FunctionConstructor.cpp

```diff
@@ -141,27 +141,27 @@ JSObject* constructFunction(JSGlobalObject* globalObject, const ArgList& args, c
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
 
+    std::optional<int> functionConstructorParametersEndPosition;
+    auto code = stringifyFunction(globalObject, args, functionName, functionConstructionMode, scope, functionConstructorParametersEndPosition);
+    EXCEPTION_ASSERT(!!scope.exception() || !code.isNull());
+
     if (UNLIKELY(!globalObject->evalEnabled())) {
-        auto codeScope = DECLARE_THROW_SCOPE(vm);
-        std::optional<int> functionConstructorParametersEndPosition;
-        auto code = stringifyFunction(globalObject, args, functionName, functionConstructionMode, codeScope, functionConstructorParametersEndPosition);
+        scope.clearException();
         globalObject->globalObjectMethodTable()->reportViolationForUnsafeEval(globalObject, !code.isNull() ? jsNontrivialString(vm, WTFMove(code)) : nullptr);
         throwException(globalObject, scope, createEvalError(globalObject, globalObject->evalDisabledErrorMessage()));
         return nullptr;
     }
-    RELEASE_AND_RETURN(scope, constructFunctionSkippingEvalEnabledCheck(globalObject, args, functionName, sourceOrigin, sourceURL, taintedOrigin, position, -1, functionConstructionMode, newTarget));
+    if (UNLIKELY(code.isNull()))
```

Comment:
You'll want an `EXCEPTION_ASSERT(!scope.exception());` here to placate the checker because the checker does not know that `globalObject->evalEnabled()` will always return false when an exception occurs and will handle it in the if block above.

Reply:
I meant above this line.

Reply:
Nevermind, Alexey explained offline that the exception source comes from `stringifyFunction`.  `globalObject->evalEnabled` cannot throw.  So, this is not needed.

However, that EXCEPTION_ASSERT should be:
```
EXCEPTION_ASSERT(!!scope.exception() || !globalObject->evalEnabled());
```
... because that is the condition you checked.

Reply:
Talked to Mark offline, we can clarify & simplify exception checking to `EXCEPTION_ASSERT(!!scope.exception() == code.isNull())`.